oauthenticator/openshift.py

```diff
@@ -190,6 +200,16 @@ async def _add_openshift_group_info(self, user_info: dict):
             self.log.warning(msg)
             return None
 
+    async def refresh_user(self, user, handler=None):
+        # Retrieve user authentication info, decode, and check if refresh is needed
+        auth_state = await user.get_auth_state()
+
+        user_info = await self._get_openshift_user_info(auth_state['access_token'])
+        if not user_info:
+            await user.stop()
+
+        return user_info
+
```

Comment:
@vpavlin who calls refresh_user? seems like this function returns user_info even when None?

Reply:
JupyterHub itself calls it regularly based on `auth_refresh_age` configuration...default is `300s`.

It also gets called before spawning new server when `refresh_pre_spawn=True` which is desired for us

It returns 3 states `True` (user info is still valid), `False` (user info is invalid and needs re-authentication), `dict` (user info needs to be updated with given values).

So when we return `None` it means user needs to go through auth again.